docusaurus/docs/dev-docs/plugins/documentation.md

```diff
@@ -51,332 +52,216 @@ This plugin comes with an interface that is available in your administration pan
 
-As you can see in that example, you are defining "no security" whereas normally you would need a bearer token to access. You will need to do this manually as the documentation plugin rewrites files and cannot pull permissions from the database as this would require a server restart each time the docs are updated.
+**`registerOverride()`**
 
-### `swagger.json` and `openapi.json` files
+| Parameter                     | Type                      | Description                                                                                                   |
+| ----------------------------- | ------------------------- | ------------------------------------------------------------------------------------------------------------- |
+| override                      | Object                    | OpenAPI object including any of the following keys paths, tags, components. Accepts JavaScript, JSON, or yaml |
+| options                       | Object                    | Accepts pluginOrigin and excludeFromGeneration                                                                |
+| options.pluginOrigin          | String                    | The plugin that is registering the override                                                                   |
+| options.excludeFromGeneration | String or Array of String | The name of the api/plugin, or list of names, to exclude                                                      |
 
-This plugin is able to serve up these files. You need to grant the permission to call the controller action to the roles that can have access.
-
-If the default `/documentation` base URL is used, the endpoints exposed are:
+:::note
+Plugin developers providing an override should always specify the `pluginOrigin` options key, otherwise the override will run regardless of the user’s config.
+:::
 
-- `/documentation/swagger.json` or `/documentation/openapi.json`: the OpenAPI spec for the latest version of your API. Both endpoints serve the same
-  document.
-- `/documentation/vN.N.N/swagger.json` or `/documentation/vN.N.N/openapi.json`: request the spec for a specific version of your API. For example: `/documentation/v1.0.0/openapi.json`
+The documentation plugin will use the registered overrides to replace the value of common keys on the generated documentation with what the override provides. If no common keys are found it will just add the new keys to the generated documentation.
 
-## FAQ
+If the override completely replaces what the documentation generates then you can specify that generation is no longer necessary by providing the names of the apis or plugins to exclude in the options key array `excludeFromGeneration`.
 
-### How does it generate the other plugins' documentation?
+If the override should only be applied to a specific version then the override must include a value for `info.version` , otherwise the override will run on all documentation versions.
```

Comment:
```suggestion
If the override should only be applied to a specific version, the override must include a value for `info.version`. Otherwise, the override will run on all documentation versions.
```